### PR TITLE
[OTTO-253] Proposal: describe "retired" versions of PHP

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Closes #n/a

## Effect
This PR is a *proposal* for a new policy on PHP versions.

The new version of wp-cli deployed to the platform (2.4.0) does not work with PHP version 5.3. This causes problems for WordPress sites running on old versions of PHP. See [OTTO-253](https://getpantheon.atlassian.net/browse/OTTO-253) for details.

One possible solution is proposed here: we could declare PHP 5.3 as "under retirement" and require old sites to upgrade before doing any further development on the site. These sites would still continue to serve pages using the older version of PHP.

## Remaining Work
- [x] Policy review by Product
- [x] Technical review
- [ ] Copy Review

## Post Launch
**Do not remove** - To be completed by the docs team upon merge:
- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5559 | Difference |
| --- | --- | --- | --- |
| Performance Score | `51` | `42` | `-9` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5559` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18866/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18866/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by greg-1-anderson** [comment]
Note that one issue with this proposal is that affected sites might need to do a software update before upgrading PHP; however, cache clear operations won't work until PHP is upgraded.

This can be resolved in a couple of ways:

- Upgrade the site software (cache cannot be cleared, site might not work)
- Upgrade PHP
- Clear the cache

Or:

- Upgrade PHP (site page loading may fail due to PHP incompatibilities)
- Upgrade software
- Clear cache

In other words, the user might encounter some transient problems, regardless of what order the operations are performed in. However, these problems do not prevent the user from fixing the issues causing them.

**Comment by greg-1-anderson** [comment]
Note also that this policy also applies to 169 Drupal sites still on PHP 5.3, whereas in fact those sites will not experience any problems. I still think it better to make the policy applicable to PHP 5.3 regardless of framework, and over-deliver for Drupal customers.

**Comment by alexfornuto** [comment]
@greg-1-anderson Thanks for getting the ball rolling on this. I have one main concern:

Documentation should be evergreen. Documenting things that "are happening" is dangerous, and should only be done with a strong roadmap and timeline defined. 

Just my two cents. Defer to the team for additional thoughts.

File: source/content/php-versions.md [hunk 1]
```diff
@@ -26,12 +26,14 @@ The recommended PHP versions available on Pantheon are:
...
-- [7.1](https://v71-php-info.pantheonsite.io/)
 
 Click on the links above to see the complete PHP info for each version, including the list of supported PHP extensions.
 
 ### EOL PHP Versions
-Pantheon also makes PHP [7.0](https://v70-php-info.pantheonsite.io/), [5.6](https://v56-php-info.pantheonsite.io/), [5.5](https://v55-php-info.pantheonsite.io/), and [5.3](https://v53-php-info.pantheonsite.io/) available on the platform, although these are end-of-life (**EOL**), and should not be used unless absolutely necessary.
+Pantheon also makes PHP [7.1](https://v71-php-info.pantheonsite.io/), PHP [7.0](https://v70-php-info.pantheonsite.io/), [5.6](https://v56-php-info.pantheonsite.io/), and [5.5](https://v55-php-info.pantheonsite.io/) available on the platform, although these are end-of-life (**EOL**), and should not be used unless absolutely necessary. 
+
+### PHP Versions Being Retired
+PHP [5.3](https://v53-php-info.pantheonsite.io/) is being retired from the platform. Sites using PHP 5.3 will continue to serve pages, but new development cannot be done. To resume development on a site using a retired version of PHP, first upgrade the PHP version on the development site.
```
Comment (alexfornuto) [inline source/content/php-versions.md]: To follow up with my comment in the main PR comment thread, my concern is this section being time-specific. I would suggest announcing the retirement of PHP 5.3 via the Changelog, blog, and any other content format that's designed to be consumed when published. Then this section can be removed, and the rest is an automatic +1.

Comment (greg-1-anderson) [reply source/content/php-versions.md]: Actually, what I think is needed here is just a better description of the status.

I intended "being retired" to be a new state for software to be in, like the "EOL" state.

A handy comparison table:

State               | Upgrade ASAP | Site still works | Development still works
------------- | --------------- | --------------- | ------------------------
EOL                 | Yes                   | Yes                    | Yes
Being retired   | Yes                  | Yes                     | *No*

I intended "being retired" to be a perpetual state with no particular end date, i.e. we might never remove this version from the platform, we just won't let you do development with it any longer.

I would prefer it if the documentation described this new "less supported" state for a while (indeterminate duration, until Product decides to do something else); in that respect, I would prefer to change the adjective from "being retired" to something else rather than remove 5.3 support with nothing more than changelog documentation, if there is a better way to describe it.

Comment (alexfornuto) [reply source/content/php-versions.md]: Thanks for the additional context. That makes a heaping spoonful of sense. I also think that, like in your explanation above, a table is the optimal way to display what versions are in what state, with a definition list or popovers to define the various states.

**Comment by alexfornuto** [comment]
So I've moved it all around to use a table and definition list. How do we feel about that? Also, is this blocked by OTT-253, or does it close it?

CC @greg-1-anderson and @carolyn-shannon.

File: source/content/php-versions.md [hunk 2]
```diff
@@ -21,30 +24,61 @@ Changes made to the `pantheon.yml` file on a branch **are not** detected when cr
...
+
+End-of-life (**EOL**) versions are available on the platform but no longer under active development, and should not be used unless absolutely necessary.
+
+</dd>
+
+<dt>Retired</dt>
+
+<dd>
+
+Sites using retired versions of PHP will continue to serve pages, but new development cannot be done. To resume development on a site using a retired version of PHP, first upgrade the PHP version on the development site.
```
Comment (greg-1-anderson) [inline source/content/php-versions.md:62]: Maybe "on the development environment" would be clearer than "devopment site".

**Comment by greg-1-anderson** [comment]
I am a strong +1 on this PR. We just need Product to approve this as an acceptable strategy / policy. If this PR is merged, it would close OTTO-253.

File: source/content/php-versions.md (hunk 2 above)
Comment (alexfornuto) [reply source/content/php-versions.md]: 86c62ecaa761c568c16524ff3fd9d82604f07bb7

**Comment by alexfornuto** [comment]
Blocked/Hold pending product approval / OTTO-253 update.

**Comment by carolyn-shannon** [comment]
No activity on the OTTO-253 issue, so closing this for now.

**Comment by tylerdigital** [comment]
Reopening the PR, +1 from Product. Please merge at will. I've commented on OTTO-253 as well: https://getpantheon.atlassian.net/browse/OTTO-253

**Comment by carolyn-shannon** [comment]
Approved by @tylerdigital - thanks!
